### PR TITLE
PC-986: Add tests for 'DoubleKeyMap'

### PR DESCRIPTION
**Pull request by ulfryk**

### CODE REVIEW

File: src/double-key-map.spec.ts [hunk 1]
```diff
@@ -1,13 +1,185 @@
...
+      expect(dkm).to.not.equal(dkm2);
+
+      expect(dkm2.equals(dkm)).to.be.true;
+      expect(dkm.equals(dkm2)).to.be.true;
+
+      expect(dkm2.equals(dkm3)).to.be.true;
+      expect(dkm3.equals(dkm2)).to.be.true;
+
+      expect(dkm3.equals(dkm)).to.be.true;
+      expect(dkm.equals(dkm3)).to.be.true;
```
Comment (tomecko) [inline src/double-key-map.spec.ts]: aren't lines 63-67 redundant? it doesn't add more value than lines 60-61 I think

the same applies to line 73 and line 77 (and some more)

File: src/double-key-map.spec.ts [hunk 2]
```diff
@@ -1,13 +1,185 @@
...
+    it('should return existing item wrapped in Some', () => {
+      testItems.forEach(item => {
+        expect(dkm.getByBaseKey(item.id).some()).to.equal(item);
+        expect(dkm2.getByBaseKey(item.id).some()).to.equal(item);
+        expect(dkm3.getByBaseKey(item.id).some()).to.equal(item);
+      });
+    });
+
+    it('should return None for not existing items', () => {
+      const notExisitingId = 11;
```
Comment (tomecko) [inline src/double-key-map.spec.ts]: typo

File: src/double-key-map.ts [hunk 3]
```diff
@@ -25,15 +31,19 @@ export class DoubleKeyMap<K1, K2, V> {
   }
 
   @Bind public set(key: K1, secondaryKey: K2, value: V): DoubleKeyMap<K1, K2, V> {
+    // TODO: Don't allow duplicate secondaryKey
+    // TODO: Remove any unused keys
```
Comment (tomecko) [inline src/double-key-map.ts]: maybe add tests for that (but comment them out ofc)

File: src/double-key-map.spec.ts [hunk 4]
```diff
@@ -1,13 +1,185 @@
...
+    it('should return existing item wrapped in Some', () => {
+      testItems.forEach(item => {
+        expect(dkm.getBySecondaryKey(item.foo).some()).to.equal(item);
+        expect(dkm2.getBySecondaryKey(item.foo).some()).to.equal(item);
+        expect(dkm3.getBySecondaryKey(item.foo).some()).to.equal(item);
+      });
+    });
+
+    it('should return None for not existing items', () => {
+      expect(dkm.getBySecondaryKey('qwerty-asdf-zxcvb').isNone()).to.be.true;
```
Comment (tomecko) [inline src/double-key-map.spec.ts]: consider storing 'qwerty-asdf-zxcvb' in a var

File: src/double-key-map.spec.ts (hunk 1 above)
Comment (ulfryk) [reply src/double-key-map.spec.ts]: They are meant to confirm that `equals` is commutative.

Comment (tomecko) [reply src/double-key-map.spec.ts]: lines 60-61 test it already: `dkm2.equals(dkm)` and `dkm.equals(dkm2)`

Comment (ulfryk) [reply src/double-key-map.spec.ts]: ahh, ok - other tests are to be sure that instances created for same data but in different way will also equal (constructor vs. empty().st(…).set(…) vs. constructor with reordered data).

**Pull request merged**